committee/heightmanager_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2019 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package committee
+
+import (
+	"strconv"
+	"testing"
+	"time"
+)
+
+var baseTime = time.Unix(int64(1546272000), 0) // 2019-01-01 00:00:00
+var hm *heightManager
+
+func TestNewHeightManager(t *testing.T) {
+	hm = newHeightManager()
+	if hm == nil {
+		t.Error("newHeightManager() failed")
+	}
+}
+
+// 0-1546272000
+// 1-1546272015
+// 2-1546272030
+// 3-1546272045
+// 4-1546272060
+func TestAdd(t *testing.T) {
+	type args struct {
+		height uint64
+		ts     time.Time
+	}
+	type test struct {
+		name    string
+		m       *heightManager
+		args    args
+		wantErr bool
+	}
+	tests := []test{}
+
+	for i := 0; i < 5; i++ {
+		ts := baseTime.Add(time.Second * time.Duration(15*i)) // increments in 15 seconds
+		name := strconv.Itoa(i) + "-" + strconv.FormatInt(ts.Unix(), 10)
+		te := test{name, hm, args{uint64(i), ts}, false}
+		tests = append(tests, te)
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.m.add(tt.args.height, tt.args.ts); (err != nil) != tt.wantErr {
```

Comment:
err := tt.m.add(tt.args.height, tt.args.ts)
if tt.wantErr {
  require.Error(err)
} else {
  require.NoError(err)
}

In general, we don't need to make it so complicated by defining struct.

Reply:
it's auto-generated code, actually, not very confused.